domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/genericlayout/GenericLayoutEntityValidatorImpl.java

```diff
@@ -1,68 +1,159 @@
 package uk.gov.hmcts.ccd.definition.store.domain.validation.genericlayout;
 
-import org.springframework.stereotype.Component;
-import uk.gov.hmcts.ccd.definition.store.domain.validation.SimpleValidationError;
-import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationErrorMessageCreator;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
 import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.CaseFieldEntityUtil;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseFieldEntity;
 import uk.gov.hmcts.ccd.definition.store.repository.entity.GenericLayoutEntity;
 
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
 @Component
 public class GenericLayoutEntityValidatorImpl implements GenericLayoutValidator {
```

Comment:
I would split this into multiple validators. What's the point of creating an interface  `GenericLayoutValidator` if then we have all the different validations happening in 1 subclass only?

Reply:
Done